geonotebook/vis/ktile/handler.py

```diff
@@ -0,0 +1,168 @@
+            output = layer.getTileResponse(coord, extension)
+            return output
+
+        return layer.getTileResponse(coord, extension)
+
+
+class KtileHandler(IPythonHandler):
+
+    def check_xsrf_cookie(self):
+        # TODO: Find a way to correctly communicate XSRF secret to
+        #       the kernel so ingest requests can be property authenticated
+        pass
+
+    def initialize(self, ktile_config_manager):
+        self.ktile_config_manager = ktile_config_manager
+
+        try:
+            if self.request.headers["Content-Type"].startswith(
+                    "application/json"):
+                self.request.json = json.loads(self.request.body)
+        except Exception:
+            self.request.json = None
+
+    def post(self, kernel_id):
+        # Note:  needs paramater validation
+        kwargs = {} if self.request.json is None else self.request.json
+
+        self.ktile_config_manager.add_config(kernel_id, **kwargs)
+        self.log.info("Created config for {}".format(kernel_id))
+        self.finish()
+
+    def delete(self, kernel_id):
+        try:
+            del self.ktile_config_manager[kernel_id]
+        except KeyError:
+            raise web.HTTPError(404, u'Kernel %s not found' % kernel_id)
+
+    def get(self, kernel_id, **kwargs):
+        try:
+            config = self.ktile_config_manager[kernel_id]
+        except KeyError:
+            raise web.HTTPError(404, u'Kernel %s not found' % kernel_id)
+
+        self.finish(serialize_config(config))
+
+
+class KtileLayerHandler(IPythonHandler):
+    def check_xsrf_cookie(self):
+        # TODO: Find a way to correctly communicate XSRF secret to
+        #       the kernel so ingest requests can be property authenticated
+        pass
+
+    def initialize(self, ktile_config_manager):
+        self.ktile_config_manager = ktile_config_manager
+
+    def prepare(self):
+        try:
+            if self.request.headers["Content-Type"].startswith(
+                    "application/json"):
+                self.request.json = json.loads(self.request.body)
+        except Exception:
+            self.request.json = None
+
+    def post(self, kernel_id, layer_name):
+        # Note: needs paramater validation
+        try:
+            self.ktile_config_manager.add_layer(
+                kernel_id, layer_name, self.request.json)
+
+            self.write({'status': 1})
+
+        except Exception:
+            import sys
+            import traceback
+            t, v, tb = sys.exc_info()
+
+            self.log.error(''.join(traceback.format_exception(t, v, tb)))
+
+            self.write({'status': 0,
+                        'error': traceback.format_exception(t, v, tb)})
+
+    def get(self, kernel_id, layer_name, **kwargs):
+        try:
+            layer = self.ktile_config_manager[kernel_id].layers[layer_name]
+        except KeyError:
+            raise web.HTTPError(404, u'Kernel %s not found' % kernel_id)
```

Comment:
Since the resource is the layer, I think it should be a `404` for the layer not being found and something like `400` for the kernel not being found.